src/prpy/serialization.py

```diff
@@ -381,10 +454,19 @@ def deserialize_kinbody_state(body, data):
             )
             raise
 
-    body.SetLinkTransformations(
-        map(deserialize_transform, data['link_transforms']),
-        data['dof_branches']
-    )
+    link_transforms = data.get('link_transforms')
+    dof_branches = data.get('dof_branches')
+    if link_transforms is not None and dof_branches is not None:
+        body.SetLinkTransformations(
+            map(deserialize_transform, link_transforms),
+            dof_branches
+        )
+
+    else:
+        deserialization_logger.warn(
+            ('KinBody "{}" does not have link_transforms/dof_branches'
+            + 'saved; falling back to dof_values').format(body.GetName()))
```

Comment:
This awkward string concatenation is not necessary. Python does string concatenation like C (your favorite, @cdellin!):
```
>>> print ('banana'
           'orange')
bananaorange